src/Core/Analysis/AnalysisIssue.cs

```diff
@@ -38,6 +42,7 @@ public class AnalysisIssue : IAnalysisIssue
             EndLineOffset = endLineOffset;
             FilePath = filePath;
             Message = message;
+            Locations = locations.Reverse().ToList().AsReadOnly();
```

Comment:
Most issues don't have secondary locations so most of the time line 45 won't be doing anything except doing lots of object allocations to represent "no secondary locations".

In this case, I'd probably add an optimisation for the "normal" case to use a "empty" singleton if the supplied list of locations is null or empty.

There's a precedent for this in the `SonarQubeRule` class that defines a singleton `Empty` dictionary to avoid creating unnecessary objects.